Baby Monitor/Source Files/Services/MediaPlayer/AudioFile/AudioFileService.swift

```diff
@@ -0,0 +1,59 @@
+//
+//  AudioFileService.swift
+//  Baby Monitor
+
+import AVFoundation
+import RxSwift
+
+protocol AudioFileServiceProtocol: ErrorProducable {
+
+    /// Create, save, and upload an audio file from input audio buffer.
+    /// - Parameters:
+    ///     - buffer: A buffer which should be converted and sent.
+    ///     - audioRecordingURL: and URL to the recording folder locally.
+    ///     - filePrefixName: a perfix of the audio files name.
+    func uploadRecordingIfNeeded(from buffer: AVAudioPCMBuffer, audioRecordingURL: URL, filePrefixName: String)
+}
+
+/// Service which handles audio file creation, saving it locally and then sending it to the server.
+final class AudioFileService: AudioFileServiceProtocol {
+
+    enum AudioFileError: Error {
+        case convertionToFileFailure
+    }
+
+    var errorObservable: Observable<Error> {
+        return Observable.merge(audioFileStorage.errorObservable, audioFileStorage.errorObservable, errorPublisher.asObservable())
```

Comment:
```suggestion
        return Observable.merge(audioFileStorage.errorObservable, errorPublisher.asObservable())
```

Reply:
oh, good catch! Here actually should have been another object.